clang/test/CodeGenSYCL/esimd-accessor-ptr-md.cpp

```diff
@@ -0,0 +1,30 @@
+// RUN: %clang_cc1 -fsycl -fsycl-explicit-simd -fsycl-is-device -triple spir64-unknown-unknown-sycldevice -emit-llvm %s -o - | FileCheck %s
+
+// This test checks proper metadata generation for accessors used in ESIMD
+// kernels:
+// 1) Proper 'kernel_arg_accessor_ptr' metadata is generated by the FE for ESIMD
+//    kernels
+// 2) Pointers originating from accessors are marked with 'buffer_t' and proper
+//    argument kind.
+
+#include "Inputs/sycl.hpp"
+
+void test(int val) {
+  cl::sycl::accessor<int, 1, cl::sycl::access::mode::read_write> accessorA;
+  cl::sycl::accessor<int, 1, cl::sycl::access::mode::read> accessorB;
+
+  cl::sycl::kernel_single_task<class esimd_kernel>(
```

Comment:
Could you please use queue::submit and handler::single_task from the mock header instead? And include the mock header as system header using `-internal-isystem` . Like here https://github.com/intel/llvm/blob/sycl/clang/test/CodeGenSYCL/stall_enable.cpp

We need to follow single style in FE test and this style must be as close as possible to the way how we use real sycl headers.

Reply:
I don't see single style in FE tests, so I picked on. But OK, I will try to rework as you suggest.

Reply:
> I don't see single style in FE tests, so I picked on. But OK, I will try to rework as you suggest.

Yes, we don't have single style yet. This problem is already captured, and I remember that someone was working on it, but it is not finished.